app/src/main/java/org/openobservatory/ooniprobe/model/settings/Settings.java

```diff
@@ -81,9 +83,10 @@ public Annotations(Context c) {
 		public Integer all_endpoints;
 
 		public Options(Context c, PreferenceManager pm) {
-			ca_bundle_path = c.getCacheDir() + "/" + Application.CA_BUNDLE;
-			geoip_asn_path = c.getCacheDir() + "/" + Application.ASN_MMDB;
-			geoip_country_path = c.getCacheDir() + "/" + Application.COUNTRY_MMDB;
+			//TODO where to check these are not null?
+			ca_bundle_path = MKResourcesManager.getCABundlePath(c);
+			geoip_country_path = MKResourcesManager.getCountryDBPath(c);
+			geoip_asn_path = MKResourcesManager.getASNDBPath(c);
```

Comment:
Having seen the code, I believe we have the same behaviour if we just call this method when the app is loaded because that's what the pre-existing code was already doing. I encourage you to do that and then open an issue saying that in some cases we may have issues if the app is open because the user may have cancelled the cache.

I'm leaning towards making the resources loading something inside android-libs in the next release, so all this complexity is not exposed to the application and we try to load them wherever needed from android-libs.

Sounds good?

Reply:
I added the checks on every run test. check the edits in AbstractTest